src/index.js

```diff
@@ -1,41 +1,72 @@
 import './style.css';
 
+const storageData = localStorage.getItem('todoItems');
 // Array to hold todo list items
-const todoItems = [];
-let index = 0;
-
-const renderTodo = (todo) => {
-  // select ul element
-  const list = document.querySelector('.todo-list');
-
-  // check if todo is completed
-  const isCompleted = todo.completed ? 'done' : '';
-
-  // create li element
-  const li = document.createElement('li');
-  li.setAttribute('class', `todo-item ${isCompleted}`);
-  li.setAttribute('data-key', todo.index);
-
-  li.innerHTML = `
-    <input id="${todo.index}" type="checkbox"/>
-    <label for="${todo.index}" class="tick js-tick"></label>
-    <span>${todo.description}</span>
-    <i class="fa fa-trash-alt"></i>
-  `;
+let todoItems = (storageData) ? JSON.parse(localStorage.getItem('todoItems')) : [];
+
+const renderTodo = () => {
+  // local storage
 
-  list.append(li);
+  let ulHTML = '';
+  todoItems.forEach((item) => {
+    // check if todo is completed
+    const isCompleted = item.completed ? 'done' : '';
+
+    // create html for todo item
+    ulHTML += `
+    <li class="todo-item ${isCompleted}" data-key="${item.index}">
+  <input id="${item.index}" type="checkbox"/>
+  <label for="${item.index}" class="tick js-tick"></label>
+  <span class="todo-desc">${item.description}</span>
+  <button class="edit" id="editBtn" type="button">Edit</button>
+  <i class="fa fa-trash-alt delete"></i>
+  </li>
+  `;
+  });
+  // append todo items to the list
+  document.querySelector('.todo-list').innerHTML = ulHTML;
 };
 
 // Function to create new todo object
 const addTodo = (description) => {
-  const todo = {
+  // create new todo object
+  todoItems.push({
     description,
     completed: false,
-    index,
+    index: todoItems.length,
+  });
+
+  // save to local storage
+  localStorage.setItem('todoItems', JSON.stringify(todoItems));
+  renderTodo();
+};
+
+const toggleDone = (key) => {
+  const index = todoItems.findIndex((item) => item.index === Number(key));
+  todoItems[index].completed = !todoItems[index].completed;
+  renderTodo(todoItems[index]);
+};
+
+const deleteTodo = (key) => {
+  // find item in the todo items array
+  const index = todoItems.findIndex((item) => item.index === Number(key));
+
+  // create new object with a deleted property
+  const todo = {
+    deleted: true,
+    ...todoItems[index],
   };
 
```

Comment:
- [ ] As requested by the previous reviewer, Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. Please fix this issue. (The video you recorded is showing the list number of items from the local storage, Those values are always in order because that is a list of items, Please check your index values while deleting a task). 

## See the image for reference 

![image](https://user-images.githubusercontent.com/83803414/172378681-5a6487fd-39ca-43d1-a5c9-c0f4d7574a8e.png)

Reply:
Okay. Working on it.

Reply:
This issue is well resolved :heavy_check_mark: